### PR TITLE
Convert 415s from DCR to 404s

### PR DESCRIPTION
**Pull request by DavidLawes**
## What does this change?
DCR was updated to explicitly handle components not being renderable in AMP (https://github.com/guardian/dotcom-rendering/pull/3563). This resulted in DCR returning 415 error codes to frontend.

The motivation for that change was to reduce 5xx alerts being returned from the rendering load balancer. However we're still seeing 5xx alerts from the router load balancer and suspect this is because frontend is returning 500 errors to the client when DCR returns a 415.

This change updates frontend to match on 415 error codes from dcr and translate them to a 404 error code for the client. We felt a 'not found' error code was appropriate for the client (given the article can't be returned in AMP format).

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |

[before]: https://user-images.githubusercontent.com/45561419/140961788-f4abfaa4-bf9f-476a-9461-b4bb79685b26.png
[after]: https://user-images.githubusercontent.com/45561419/140961843-130ac3b8-2803-4112-934a-fe6b8bc28898.png

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: common/app/renderers/DotcomRenderingService.scala [hunk 1]
```diff
@@ -79,6 +79,12 @@ class DotcomRenderingService extends GuLogging with ResultWithPreconnectPreload
           // if DCR returns a 400 it's because *we* failed, so frontend should return a 500
           NoCache(play.api.mvc.Results.InternalServerError("Remote renderer validation error (400)"))
             .withHeaders("X-GU-Dotcomponents" -> "true")
+        case 415 =>
+          // if DCR returns a 415 it's because we can't render a specific component, so page is not available
+          NoCache(
```
Comment (nicl) [inline common/app/renderers/DotcomRenderingService.scala:84]: There is a `NotFound` default cache value we could use here: https://github.com/guardian/frontend/blob/main/common/app/model/Cached.scala#L26.

In any case, I think we need to have some caching on these.

**Review by nicl — COMMENTED** [review]
Looks great. Only blocking thing is around caching.

Also, might be worth chatting with @rcrphillips about impact (if any) of 404ing AMP pages. Are these pages Google attempts to index for AMP?

File: common/app/renderers/DotcomRenderingService.scala (hunk 1 above)
Comment (davidfurey) [reply common/app/renderers/DotcomRenderingService.scala:84]: Sorry, I forgot about the surrogate header.  You should set this header so we do for other 404s.

**Comment by DavidLawes** [comment]
We have discussed how to deploy and measure the impact of this:
- `Will it impact how Google scrapes amp articles?` With the help of @idrise I believe we can monitor the google search console after the deploy for any increase in scraping errors (I don't have access to the search console). From a brief discussion with David F last week I didn't get the feeling that mutating the error code to a non-500 was something different compared to pre-DCR days.
- `Should we restrict the change to specific IP addresses?` The original issue was that 2 specific IP addresses were spamming us for the amp version of non-renderable articles (which triggered pager duty alerts). Looking at the errors we're now receiving the IP addresses differ from those I originally recorded. Additionally, there seem to be more than 2 unique IP addresses requesting non-renderable amp content. Therefore, I don't think we can restrict this change to specific IP addresses.

We could also consider redirecting to the html version if we wanted, as a future change. For now, I believe there's benefit in this change as it will help reduce pager duty noise.

**Review by davidfurey — APPROVED** [review]

**Comment by DavidLawes** [comment]
[Convert 415s from DCR into appropriate 4xx responses to client](https://trello.com/c/W0NLg77Z/363-convert-415s-from-dcr-into-appropriate-4xx-responses-to-client)

**Pull request merged**